elasticsearch/file/src/main/java/nl/knaw/huc/service/PageTurner.java

```diff
@@ -0,0 +1,95 @@
+package nl.knaw.huc.service;
+
+import com.jayway.jsonpath.DocumentContext;
+import com.jayway.jsonpath.ParseContext;
+import com.jayway.jsonpath.TypeRef;
+import org.glassfish.jersey.client.JerseyClientBuilder;
+
+import javax.ws.rs.client.Client;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.List;
+import java.util.function.Consumer;
+
+import static nl.knaw.huc.service.ResourceUtil.read;
+
+/**
+ * Turn the pages of a paginated resource based on query params:
+ * - limit
+ * - offset (zero-based)
+ * The following json-page-structure is expected:
+ * ```
+ * {
+ * "items": [],
+ * "total": number
+ * }
+ * ```
+ *
+ * <p>T is type of item in $.items.
+ * JsonPath needs TypeRef to infer type correctly
+ */
+public class PageTurner<T> {
```

Comment:
Conceptueel kan ik snappen wat dit ding doet, zeker gegeven de goede naamgeving. Maar ik ben een beetje het spoor bijster waarom we dit ding nodig hebben (we hadden toch al Paging support?)
Hier wordt duideljk dat er gewoon een flink hiaat in mijn kennis zit rondom de indexers, vrees ik.
Is het een idee hier nog wat Java tests bij te maken, gegeven de vrij specifieke logica in turn() om eventuele off-by-ones af te dekken?
Mogelijk zit dat ergens verwerkt in een Concordion test, maar die zie ik niet in dit PR.

Reply:
Dit ding haalt alle pagina's op uit de TextRepo, en geeft de inhoud van iedere pagina door aan de `reader`. 
De TextRepo heeft pagina's, en dit is de code die de pagina's ophaalt.

Reply:
Om iets meer context te geven: de FileIndexer heeft alle versies nodig van een file: 1 pagina ophalen is dus niet genoeg.

Reply:
Ik snap. Dank!